nabs/simulators.py

```diff
@@ -0,0 +1,244 @@
+"""Simulation and validation functions for plans"""
+import itertools
+import logging
+import sys  # NOQA
+from contextlib import contextmanager
+from typing import Any, Generator, Iterator, List, Tuple
+
+from bluesky.simulators import check_limits
+
+logger = logging.getLogger(__name__)
+
+
+class ValidError(Exception):
+    pass
+
+
+def raiser(*args: Any, **kwargs: Any):
+    raise ValidError('forbidden method called')
+
+
+# Be wary of how you specify these, they are keyed based on
+# how they were imported.
+default_patches = [
+            "sys.modules['ophyd'].sim.SynAxis.set",
+            "sys.modules['ophyd'].signal.EpicsSignal.put",
```

Comment:
These seem like reasonable spots to monkeypatch. I was trying to figure out if doing something to the control layer e.g.
```
from ophyd import cl
# do something to the cl here
```
could also work but it seems annoying